src/tools/rados/rados.cc

```diff
@@ -407,20 +525,30 @@ static int do_put(IoCtx& io_ctx, RadosStriper& striper,
  
     if (count == 0) {
      if (offset == obj_offset) { // in case we have to create an empty object & if obj_offset > 0 do a hole
+
+#ifndef WITH_LIBRADOSSTRIPER
+	ret = io_ctx.write_full(oid, indata); // indata is empty
+#else
 	if (use_striper) {
 	  ret = striper.write_full(oid, indata); // indata is empty
 	} else {
 	  ret = io_ctx.write_full(oid, indata); // indata is empty
 	}
+#endif
 	if (ret < 0) {
 	  goto out;
 	}
+
 	if (offset) {
+#ifndef WITH_LIBRADOSSTRIPER
```

Comment:
This `#ifndef` appears to be a noop because it's inside `#else // WITH_LIBRADOSSTRIPER`